File: C++/Baekjoon/Problem1929.cpp
#include <iostream>
using namespace std;

void Eratos(int m, int n)
{
    bool* PrimeArray = new bool[n+1];

    for(int i=2; i<=n; i++)
        PrimeArray[i]= true; //일단 전부 true

    for(int i=2; i*i<=n; i++)
    {
        if(PrimeArray[i])
        {
            for(int j=i*i; j<=n; j+=i)
                PrimeArray[j] = false;
        }
    }
    PrimeArray[1]=false;
    for(int i=m; i<=n; i++)
    {
        if(PrimeArray[i]) cout<<i<<'\n';
    }
}

int main()
{
    ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

    int m,n;
    cin>>m>>n;

    Eratos(m,n);
}

File: Python/Baekjoon/Problem1929.py
def Eratos(m, n):
    PrimeArray = [True] * (n + 1)
    end = int(n ** (1 / 2))
    for i in range(2, end + 1):
        if PrimeArray[i] is True:
            for j in range(i * i, n + 1, i):
                PrimeArray[j] = False

    PrimeArray[1] = False
    for i in range(m, n + 1):
        if PrimeArray[i] is True:
            print(i, end='\n')

m, n = map(int, input().split())
Eratos(m, n)
